modules/usage_sample.cpp

```diff
@@ -7,10 +7,46 @@
 // clang++ -std=c++20 -fmodule-file=type_index.pcm type_index.pcm usage_sample.cpp
 
 //[any_module_example
+
+#ifdef BOOST_ANY_USE_STD_MODULE
+import std;
+#else
+# include <sstream>
+# include <string>
+# include <print>
+#endif
+
 import boost.any;
 
-int main() {
-    boost::any a = 42;
+namespace {
+
+    template <typename... Ts> auto any_to_string(const boost::any& a) -> std::string {
```

Comment:
too much for a smoke test

Previous version was shorter and simpler, without `#ifdef` boilerplate